workspace_host/interface.js

```diff
@@ -627,47 +803,56 @@ function _createContainer(workspace_id, port, settings, callback) {
         });
 }
 
-function _createContainerWrapper(workspace_id, port, settings, callback) {
+function _createContainerWrapper(workspace, callback) {
     async.parallel({
-        query: (callback) => {_queryUpdateWorkspaceHostname(workspace_id, port, callback);},
-        container: (callback) => {_createContainer(workspace_id, port, settings, callback);},
+        query: (callback) => {_queryUpdateWorkspaceHostname(workspace.id, workspace.launch_port, callback);},
+        container: (callback) => {_createContainer(workspace, callback);},
     }, (err, results) => {
         if (ERR(err, callback)) return;
-        callback(null, workspace_id, results.container);
+        workspace.container = results.container;
+        callback(null, workspace);
     });
 }
 
-function _startContainer(workspace_id, container, callback) {
-    container.start((err) => {
+function _startContainer(workspace, callback) {
+    workspace.container.start((err) => {
         if (ERR(err, callback)) return;
-        callback(null, workspace_id, container);
+        callback(null, workspace);
     });
 }
 
 // Called by the main server the first time a workspace is used by a user
 function initSequence(workspace_id, useInitialZip, res) {
     logger.info(`Launching workspace_id=${workspace_id}, useInitialZip=${useInitialZip}`);
 
-    id_workspace_mapper[workspace_id] = {};
-    id_workspace_mapper[workspace_id].localName = `workspace-${uuidv4()}`;
-    id_workspace_mapper[workspace_id].s3Name = `workspace-${workspace_id}/current`;
+    const uuid = uuidv4();
+    const workspace = {
+        'id': workspace_id,
+        'launch_uuid': uuid,
+        'local_name': `workspace-${uuid}`,
+        's3_name': `workspace-${workspace_id}`,
```

Comment:
```suggestion
        's3_name': `workspace-${workspace_id}/current`,
```
Fixes the container-s3 syncing regression.